macOS Brew recipe for docker-compose is failing to install

## Context

https://github.com/techservicesillinois/awscli-login/actions/runs/5894707628/job/16018161150?pr=149#step:12:62

Our working theory is that the `brew` recipe may no longer be compatible with GitHub actions.
It looks like it is trying to start [Qemu][1] during install, and it is failing. 
We would not expect any kind of VM to work on GitHub actions.

[1]: https://www.qemu.org/

## Tasks

- [ ] update integration tests to only run on Linux and Windows in CI/CD

## Error trace from GitHub action

```bash
Run brew install docker docker-compose
Warning: Treating docker as a formula. For the cask, use homebrew/cask/docker
==> Fetching dependencies for docker: docker-completion
==> Fetching docker-completion
==> Downloading https://ghcr.io/v2/homebrew/core/docker-completion/manifests/24.0.5
==> Downloading https://ghcr.io/v2/homebrew/core/docker-completion/blobs/sha256:b77b6d4980dd2c9345cf64b9a790c0ac177358195d19202a06112b19b1533a1d
==> Fetching docker
==> Downloading https://ghcr.io/v2/homebrew/core/docker/manifests/24.0.5
==> Downloading https://ghcr.io/v2/homebrew/core/docker/blobs/sha256:25a7049a99a9a2e8a9729a2884a177e5fb692f160efd07db5be26a44d5174c66
==> Fetching docker-compose
==> Downloading https://ghcr.io/v2/homebrew/core/docker-compose/manifests/2.20.3
==> Downloading https://ghcr.io/v2/homebrew/core/docker-compose/blobs/sha256:174117b0c540c90033c0835e493f19bf96d5bc97a28dd12f0b2a19e179aa8638
==> Installing dependencies for docker: docker-completion
==> Installing docker dependency: docker-completion
==> Pouring docker-completion--24.0.5.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/docker-completion/24.0.5: 9 files, 337.8KB
==> Installing docker
==> Pouring docker--24.0.5.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/docker/24.0.5: 13 files, 32.2MB
==> Pouring docker-compose--2.20.3.monterey.bottle.tar.gz
==> Caveats
Compose is now a Docker plugin. For Docker to find this plugin, symlink it:
  mkdir -p ~/.docker/cli-plugins
  ln -sfn /usr/local/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
==> Summary
🍺  /usr/local/Cellar/docker-compose/2.20.3: 6 files, 57.5MB
==> Caveats
==> docker-compose
Compose is now a Docker plugin. For Docker to find this plugin, symlink it:
  mkdir -p ~/.docker/cli-plugins
time="2023-08-18T16:37:03Z" level=info msg="starting colima"
  ln -sfn /usr/local/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
time="2023-08-18T16:37:03Z" level=info msg="runtime: docker"
time="2023-08-18T16:37:03Z" level=info msg="preparing network ..." context=vm
time="2023-08-18T16:37:03Z" level=info msg="creating and starting ..." context=vm
time="2023-08-18T16:37:03Z" level=info msg="Terminal is not available, proceeding without opening an editor"
time="2023-08-18T16:37:03Z" level=info msg="Attempting to download the image" arch=x86_64 digest="sha512:f761b807fe9ba345968df72c07f8c5abcae0c4a44976fe5595c0ff748ef693841221a70e663986c700b027cea32b7cac24d5490d4c721593c39f2b8840c362a2" location="https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.5/alpine-lima-clm-3.18.0-x86_64.iso"
Downloading the image (alpine-lima-clm-3.18.0-x86_64.iso)

315.00 MiB / 315.00 MiB (100.00%) ? p/stime="2023-08-18T16:37:05Z" level=info msg="Downloaded the image from \"[https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.5/alpine-lima-clm-3.18.0-x86_64.iso\](https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.5/alpine-lima-clm-3.18.0-x86_64.iso/)""
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] Starting QEMU (hint: to watch the boot progress, see \"/Users/runner/.lima/colima/serial*.log\")"
time="2023-08-18T16:37:09Z" level=info msg="SSH Local Port: 49228"
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] Waiting for the essential requirement 1 of 5: \"ssh\""
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] Driver stopped due to error: \"exit status 255\""
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] Shutting down the host agent"
time="2023-08-18T16:37:09Z" level=warning msg="[hostagent] failed to exit SSH master"
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] Shutting down QEMU with ACPI"
time="2023-08-18T16:37:09Z" level=warning msg="[hostagent] failed to open the QMP socket \"/Users/runner/.lima/colima/qmp.sock\", forcibly killing QEMU"
time="2023-08-18T16:37:09Z" level=info msg="[hostagent] QEMU has already exited"
time="2023-08-18T16:37:09Z" level=fatal msg="exiting, status={Running:false Degraded:false Exiting:true Errors:[] SSHLocalPort:0} (hint: see \"/Users/runner/.lima/colima/ha.stderr.log\")"
time="2023-08-18T16:37:09Z" level=fatal msg="error starting vm: error at 'creating and starting': exit status 1"
Error: Process completed with exit code 1.
```

--- a/.github/workflows/cicd.yml
+++ b/.github/workflows/cicd.yml
@@ -53,7 +53,7 @@ jobs:
             dist
             build
 
-  test:
+  unit_tests:
     runs-on: ${{ matrix.os }}
     needs: build
 
@@ -64,8 +64,6 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-      with:
-        submodules: recursive
 
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v4
@@ -123,12 +121,6 @@ jobs:
       run: |
         sudo apt-get install -y libxml2-dev libxslt-dev
 
-    - name: Install system dependencies
-      if: matrix.os == 'macos-latest'
-      run: |
-        brew install docker docker-compose
-        colima start
-
     - name: Upgrade pip
       run: |
         python -m pip install --upgrade --upgrade-strategy eager pip
@@ -154,27 +146,75 @@ jobs:
     - name: Coverage report
       run: make coverage report
 
-    - name: Install (local)
-      run: make .install
-
-    - name: Integration Tests (POSIX)
-      if: matrix.os != 'windows-latest'
-      run: make integration_tests
-
-    - name: Integration Tests (Windows)
-      if: matrix.os == 'windows-latest'
-      run: make integration_tests
-      env:
-        AWSCLI_LOGIN_WINDOWS_TEST: 1
-
     - name: Archive coverage report
       uses: actions/upload-artifact@v3
       with:
         name: coverage-${{ runner.os }}-py${{ matrix.python-version }}
         path: htmlcov
 
+  integration_tests:
+    runs-on: ${{ matrix.os }}
+
+
+    needs: build
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ["3.7","3.8","3.9","3.10","3.11"]
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive  # Needed by BATS a bash testing framework
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Download build artifacts
+      uses: actions/download-artifact@v3
+      with:
+        name: build
+
+    - name: Get pip cache dir
+      id: pip-cache
+      run: echo "::set-output name=dir::$(python -m pip cache dir)"
+
+    - name: Save pip cache
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip-${{ secrets.CACHE_VERSION }}
+
+    - name: Install system dependencies
+      if: matrix.os == 'ubuntu-latest' && steps.cache.outputs.valid != 'true'
+      run: |
+        sudo apt-get install -y libxml2-dev libxslt-dev
+
+    - name: Upgrade pip
+      run: |
+        python -m pip install --upgrade --upgrade-strategy eager pip
+
+    - name: Install build
+      run: make install-build
+
+    - name: Install integration test deps
+      run: make deps-integration-test
+
+    - name: Integration Tests (POSIX)
+      if: matrix.os != 'windows-latest'
+      run: make integration-tests
+
+    - name: Integration Tests (Windows)
+      if: matrix.os == 'windows-latest'
+      run: make integration-tests
+      env:
+        AWSCLI_LOGIN_WINDOWS_TEST: 1
+
   publish:
-    needs: test
+    needs: [unit_tests, integration_tests]
     runs-on: ubuntu-latest
 
     steps:

--- a/.gitignore
+++ b/.gitignore
@@ -24,3 +24,4 @@ docs/src/
 .twinecheck
 .vscode 
 .win-tox
+.install-build

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ PIP = python -m pip install --upgrade --upgrade-strategy eager
 
 .PHONY: all check install test lint static develop develop-coverage
 .PHONY: freeze shell clean docs coverage doctest win-tox
+.PHONY: install-build
 
 all: test coverage docs doctest
 
@@ -34,6 +35,10 @@ deps-doc:
 # Python packages needed to run tests
 deps-test:
 	$(PIP) coverage
+
+# Python packages needed to run integration_tests tests
+deps-integration-test:
+	$(PIP) vcrpy
 
 # Python packages needed to publish a production or test release
 deps-publish:
@@ -95,12 +100,16 @@ idp-down:
 	docker-compose down --rmi all
 	rm -f .idp.docker
 
+.install-build: build
+install-build: .install-build
+	pip install dist/awscli_login-*.whl
+
 ifeq ($(RUNNER_OS),Windows)
     idp_integration_deps=
 else
     idp_integration_deps=.idp.docker
 endif
-integration_tests: .install $(idp_integration_deps)
+integration-tests: $(idp_integration_deps)
 	make -C src/integration_tests/
 
 # Run tests with coverage tool -- generates .coverage file
@@ -174,7 +183,7 @@ release: .release
 	twine upload "$(RELEASE)"
 
 clean: idp-down
-	rm -rf .coverage .coverage.develop .lint .mypy_cache .static .tox .wheel htmlcov .twinecheck
+	rm -rf .coverage .coverage.develop .lint .mypy_cache .static .tox .wheel htmlcov .twinecheck .install-build
 	rm -rf $(PKG)/__pycache__ $(TPKG)/__pycache__ $(TPKG)/cli/__pycache__/ $(TPKG)/config/__pycache__
 	rm -rf build dist src/*.egg-info .eggs
 	make -C docs clean

